pyfr/integrators/dual/pseudo/multip.py

```diff
@@ -278,16 +282,21 @@ def _mg_regidx(self):
 
     def pseudo_advance(self, tcurr):
         # Multigrid levels and step counts
-        cycle, csteps = self.cycle, self.csteps
+        cycle, cstepsf = self.cycle, self.csteps
 
         # Set time step and current stepper coefficients for all levels
         for l in self.levels:
             self.pintgs[l]._dt = self._dt
             self.pintgs[l].stepper_coeffs = self.stepper_coeffs
 
         self.tcurr = tcurr
+        fgen = self._fgen
 
         for i in range(self._maxniters):
+            # Choose either ⌊c⌋ and ⌈c⌉ in a way that the average is c
+            csteps = [int(self._fgen.choice([np.floor(c), np.ceil(c)], 
+                          p=[1 - c % 1, c % 1])) for c in cstepsf]
```

Comment:
Would this be cleaner?
```
    csteps = [int(np.floor(c) if fgen.random() < 1 - c % 1 else np.ceil(c)) for c in cstepsf]
```

Reply:
Ok. Rearranging a bit:
```
csteps = [int(np.floor(c) if fgen.random() > c % 1 else np.ceil(c)) for c in cstepsf]
```

Or the below line, which runs 2 times faster and fits in the 80 lines limit, but lesser readable.
```
csteps = [int(c + (fgen.random() < c % 1)) for c in cstepsf]
```

Reply:
I think we can use the second one. I have anyways commented above that line what the line does.